### PR TITLE
Add friendly dygraph trace API

### PR DESCRIPTION
**Pull request by sneaxiy**
This PR add `TracedLayer` interface to convert the dygraph model to static graph model.

Would expose this API and add doc in the next PR.

### CODE REVIEW

File: python/paddle/fluid/dygraph/jit.py [hunk 1]
```diff
@@ -95,7 +100,7 @@ def forward(self, input):
                 layer = ExampleLayer("example_layer")
                 in_np = np.random.random([2, 3]).astype('float32')
                 in_var = to_variable(in_np)
-                out, program = jit.trace(layer, inputs=[in_var],
+                out, program, _, _ = jit._trace(layer, inputs=[in_var],
```
Comment (JiabinYang) [inline python/paddle/fluid/dygraph/jit.py:103]: better not show a private interface to user?

**Review by JiabinYang — COMMENTED** [review]
just one comment

File: python/paddle/fluid/dygraph/jit.py (hunk 1 above)
Comment (sneaxiy) [reply python/paddle/fluid/dygraph/jit.py:103]: Yes, I agree. `_trace` is not inside `__all__` of `jit.py`. So it is not a public interface. The only public interface is `TracedLayer`.

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**